pkg/boot/menu/menu.go

```diff
@@ -78,27 +93,75 @@ func Choose(input *os.File, entries ...Entry) Entry {
 		}
 
 		for {
+			term.SetPrompt("Enter an option ('01' is the default, 'e' to edit kernel cmdline):\n > ")
 			choice, err := term.ReadLine()
 			if err != nil {
 				if err != io.EOF {
-					fmt.Printf("BUG: Please report: Terminal read error: %v.\r\n", err)
+					fmt.Printf("BUG: Please report: Terminal read error: %v.\n", err)
 				}
 				boot <- nil
 				return
 			}
 
+			if choice == "e" {
+				// Edit command line.
+				term.SetPrompt("Select a boot option to edit:\n > ")
+				choice, err := term.ReadLine()
+				if err != nil {
+					fmt.Println(err)
+					continue
+				}
+				num, err := parseBootNum(choice, entries)
+				if err != nil {
+					fmt.Println(err)
+					continue
+				}
+				entries[num-1].Edit(func(cmdline string) string {
```

Comment:
check that num-1 exists first, else this panics

Reply:
This is checked in the `parseBootNum` function.